IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -57,8 +79,16 @@ + (NSString *_Nullable)getWPJStringDataForIdentifier:(nonnull NSString *)identif
 
         return nil;
     }
-
-    NSString *stringData = [(__bridge NSDictionary *)result objectForKey:(__bridge id)(kSecAttrService)];
+    NSString *stringData;
+    if (tenantId && key)
+    {
+        CFStringRef unformattedKey = (__bridge CFStringRef)key;
```

Comment:
Since, we want to pass argument as NSString for keychain query key is of type CFStringRef: 
- in the method which decides which key to query on, we need to use type CFStringRef, but need to pass it as NSString, so we format it to NSString and pass
- in the method of actually querying, we've received NSString but want to query as CFStringRef, so we unformat it back to CFStringRef.
Using the syntax for conversions from [here](https://stackoverflow.com/questions/17227348/nsstring-to-cfstringref-and-cfstringref-to-nsstring-in-arc)